File: 백준/Bronze/15829. Hashing/Hashing.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws Exception{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        int num = Integer.parseInt(br.readLine());
        String str = br.readLine();
        long answer = 0;

        for (int i = 0; i < num; i++) {
            char n = str.charAt(i);
            answer += ((int)n-96) * Math.pow(31, i);
        }

        System.out.println(answer);
    }
}

File: 백준/Bronze/15829. Hashing/Hashing.js
let path = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
let input = require('fs').readFileSync(path).toString().replace(/\r/g, '').trim().split('\n');

let loop = parseInt(input[0]);
let str = input[1];
let answer = 0;

for (let i=0; i<loop; i++) {
    answer += (str.charAt(i).charCodeAt()-96) * Math.pow(31, i);
}

console.log(answer);